src/uptane/legacysecondary.cc

```diff
@@ -7,70 +7,34 @@
 #include "logger.h"
 
 namespace Uptane {
-LegacySecondary::LegacySecondary(const LegacySecondaryConfig &sconfig_in, Uptane::Repository *primary)
-    : sconfig(sconfig_in), transport(primary), wait_image(false) {
+LegacySecondary::LegacySecondary(const LegacySecondaryConfig &sconfig_in) {
   boost::filesystem::create_directories(sconfig.firmware_path.parent_path());
 }
 
-bool writeImage(const uint8_t *blob, size_t size) {
-  if (!wait_for_target) return false;
+bool LegacySecondary::storeFirmware(const std::string& target_name, const std::string& content) {
+  // reading target hash back is not currently supported, so primary needs to save the firmware file locally
+  Utils::writeFile(sconfig.target_name_path.string(), expected_target_name);
+  Utils::writeFile(sconfig.firmware_path.string(), content);
 
-  wait_for_target = false;
-  if (size > expected_targets_length) {
-    detected_attack = "overflow";
-    return true;
-  }
-  std::string content = std::string(blob, size);
-  std::vector<Hash>::const_iterator it;
-  for (it = expected_target_hashes.begin(); it != expected_target_hashes.end(); it++) {
-    if (it->TypeString() == "sha256") {
-      if (boost::algorithm::to_lower_copy(boost::algorithm::hex(Crypto::sha256digest(content) != it->HashString()))) {
-        detected_attack = "wrong_hash";
-        return true;
-      }
-    } else if (it->TypeString() == "sha512") {
-      if (boost::algorithm::to_lower_copy(boost::algorithm::hex(Crypto::sha512digest(content) != it->HashString()))) {
-        detected_attack = "wrong_hash";
-        return true;
-      }
-    }
-  }
-  detected_attack = "";
-  Utils::writeFile(sconfig.target_name_path.string(), content);
-  return true;
+  std::string output
+  int rs = Utils::shell(sconfig.flasher.string() + " --hardware-identifier " + sconfig.ecu_hardware_id + " --ecu-identifier " + sconfig.ecu_serial + " --firmware " + sconfig.firmware_path.string(), &output);
+
+  if(rs != 0)
```

Comment:
It is better always use {}. If someone will need to add some code inside if statement, he could not notice that there is no {} symbols.